dw-ripple.js

```diff
@@ -54,17 +94,153 @@ export class DwRipple extends Ripple {
       /**
        * Set to `true` if ripple should not be shown
        */
-      disabled: { type: Boolean, reflect: true }
+      disabled: { type: Boolean, reflect: true },
+
+      /**
+       * Set to true when ripple shows as an unbounded.
+       */
+      unbounded: { type: Boolean, reflect: true }
     }
   }
 
   constructor(){
     super();
-    this.primary=false;
-    this.secondary = false;
     this.disabled = false;
+    this.unbounded = false;
+    this.__activeRipple = this.__activeRipple.bind(this);
+    this.__inactiveRipple = this.__inactiveRipple.bind(this);
+
+    /**
+     * Ripple animation is completed or not.
+     */
+    this.__isRippleComplete;
+
+    /**
+     * Ripple in-active events handler is called or not.
+     */
+    this.__removeRipple;
+  }
+
+
+  connectedCallback() {
+    super.connectedCallback && super.connectedCallback();
+    this.updateComplete.then(() => {
+      this.__bindActiveEvents();
+      this.__bindInactiveEvents();
+    });
+  }
+
+  disconnectedCallback() {
+    this.__unbindActiveEvents();
+    this.__unbindInactiveEvents();
+    super.disconnectedCallback && super.disconnectedCallback();
+  }
+
+  render() {
+    return html`
+    `;
+  }
+
+
+  /**
+   * Bind active ripple events.
+   * @private
+   */
+  __bindActiveEvents() {
+    this.parentNode && this.parentNode.addEventListener('mousedown', this.__activeRipple);
+    this.parentNode && this.parentNode.addEventListener('touchstart', this.__activeRipple);
+  }
+
+  /**
+   * unbind active ripple events.
+   * @private
+   */
+  __unbindActiveEvents() {
+    this.parentNode && this.parentNode.removeEventListener('mousedown', this.__activeRipple);
+    this.parentNode && this.parentNode.removeEventListener('touchstart', this.__activeRipple);
+  }
+
+  /**
+   * Bind remove/in-active ripple events.
+   * @private
+   */
+  __bindInactiveEvents() {
+    this.parentNode && this.parentNode.addEventListener('mouseup', this.__inactiveRipple);
+    this.parentNode && this.parentNode.addEventListener('mouseleave', this.__inactiveRipple);
+    this.parentNode && this.parentNode.addEventListener('touchend', this.__inactiveRipple);
+  }
+
+  /**
+   * unbind remove/in-active ripple events.
+   * @private
+   */
+  __unbindInactiveEvents() {
+    this.parentNode && this.parentNode.removeEventListener('mouseup', this.__inactiveRipple);
+    this.parentNode && this.parentNode.removeEventListener('mouseleave', this.__inactiveRipple);
+    this.parentNode && this.parentNode.removeEventListener('touchend', this.__inactiveRipple);
   }
 
+  /**
+   * Invoked on ripple in-active events.
+   * @private
+   */
+  __inactiveRipple() {
```

Comment:
We should remove this function and only action handler function `__fadeOut` is enough to do the work.